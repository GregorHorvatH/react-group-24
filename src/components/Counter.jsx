const CounterContext = ({ value = 0, onIncrement, onDecrement, onDelete }) => {
  return (
    <div className="counter">
      <button onClick={onDecrement}>-</button>
      <span>{value}</span>
      <button onClick={onIncrement}>+</button>

      <button onClick={onDelete}>X</button>
    </div>
  );
};

export default CounterContext;
