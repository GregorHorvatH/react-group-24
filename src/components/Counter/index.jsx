const CounterContext = ({ value = 0, onIncrement, onDecrement, onDelete }) => {
  return (
    <div className="counter">
      <button data-testid="decrement" onClick={onDecrement}>
        -
      </button>
      <span data-testid="value">{value}</span>
      <button data-testid="increment" onClick={onIncrement}>
        +
      </button>

      <button data-testid="delete" onClick={onDelete}>
        X
      </button>
    </div>
  );
};

export default CounterContext;

export const sum = (a, b) => a + b; // 1 + 2 = 3
