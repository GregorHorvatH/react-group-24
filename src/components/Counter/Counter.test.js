import { render } from '@testing-library/react';
import Counter, { sum } from './index';

describe('sum should not blow up', () => {
  it('sum(1, 2) should be 3', () => {
    expect(sum(1, 2)).toBe(3);
  });

  it('sum(-1, 2) should be 1', () => {
    expect(sum(-1, 2)).toBe(1);
  });

  it('sum(0, 0) should be 0', () => {
    expect(sum(0, 0)).toBe(0);
  });
});

describe('Counter', () => {
  it('Counter should render properly', () => {
    const { container } = render(<Counter value={10} />);

    expect(container).toMatchSnapshot();
  });

  it('Button decrement should call onDecrement', () => {
    const onDecrement = jest.fn();
    const { getByTestId } = render(
      <Counter value={10} onDecrement={onDecrement} />,
    );

    getByTestId('decrement').click();

    expect(onDecrement).toBeCalled();
  });

  it('Button increment should call onIncrement', () => {
    const onIncrement = jest.fn();
    const { getByTestId } = render(
      <Counter value={10} onIncrement={onIncrement} />,
    );

    getByTestId('increment').click();

    expect(onIncrement).toBeCalled();
  });

  it('Button delete should call onDelete', () => {
    const onDelete = jest.fn();
    const { getByTestId } = render(<Counter value={10} onDelete={onDelete} />);

    getByTestId('delete').click();

    expect(onDelete).toBeCalled();
  });

  it('The default value should be 0', () => {
    const { getByTestId } = render(<Counter />);

    const value = getByTestId('value').textContent;

    expect(value).toBe('0');
  });

  it('value should be 10', () => {
    const { getByTestId } = render(<Counter value={10} />);

    const value = getByTestId('value').textContent;

    expect(value).toBe('10');
  });
});
