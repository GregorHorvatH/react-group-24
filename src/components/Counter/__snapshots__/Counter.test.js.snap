// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Counter should render properly 1`] = `
<div>
  <div
    class="counter"
  >
    <button
      data-testid="decrement"
    >
      -
    </button>
    <span
      data-testid="value"
    >
      10
    </span>
    <button
      data-testid="increment"
    >
      +
    </button>
    <button
      data-testid="delete"
    >
      X
    </button>
  </div>
</div>
`;
